src/Factory.php

```diff
@@ -31,6 +31,19 @@ public function publish(string $broker = null): MessageProducer
         );
     }
 
+    /** Creates a new ProducerBuilder instance, setting brokers and topic. The producer will be flushed only when the application terminates, and doing SEND does not mean that the message was flushed! */
+    public function asyncPublish(string $broker = null): MessageProducer
```

Comment:
Alternatively a second argument could be introduced to the publish method: bool $async = false with the same effect but w/out the duplicated code in the factory...

Reply:
I prefer to have a dedicated method for this